### PR TITLE
Migrated PR checks to GitHub actions

### PR DESCRIPTION
**Pull request by ericmjl**
Tagging @alokito.

Partly a protest against Travis CI for instituting build time limits for even open source projects, and partly because GitHub Actions do have a few advantages:

1. A tad more organization
2. Parallel builds, unlimited build time.

I've gotten a bit of practice on [jax-unirep](https://github.com/ElArkk/jax-unirep/pull/97) and [Network Analysis Made Simple](https://github.com/ericmjl/Network-Analysis-Made-Simple/actions/runs/445659815), and I think you'll enjoy it too.

Apologies about PR-ing from `master`, my fingers moved a tad too fast for myself. I'll be deleting my fork if this PR is merged or closed.

### CODE REVIEW

File: .github/workflows/pr-checks.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        run: |
+          conda env create -f environment.yml
+          conda activate cellxgene-gateway
+          python setup.py install
+
+      - name: Run tests
+        run: |
+          conda activate cellxgene-gateway
+          python -m unittest discover tests
+          bash <(curl -s https://codecov.io/bash)
```
Comment (alokito) [inline .github/workflows/pr-checks.yaml:47]: Piping stuff from the internet directly to bash still freaks me out a little.

**Comment by alokito** [comment]
@ericmjl Quick question: I notice that you have this triggered on pull_request, but not on push, whereas Travis was building on every branch whether there was a PR or not. Is there a reason not to change the events to `on: [push, pull_request]`?

**Comment by ericmjl** [comment]
@alokito No good reason! I was just limiting scope to keep things simple. Can definitely do it on every push and pull request, which might be a great idea :smile:.

File: .github/workflows/pr-checks.yaml (hunk 1 above)
Comment (ericmjl) [reply .github/workflows/pr-checks.yaml:47]: Haha yeah. Thankfully this is all isolated in GitHub's machines. There is a CodeCov action btw, which can be used in lieu of the piped in bash script.

**Pull request merged**